### ISSUE TITLE
Auto-completion initializing nullable local creates invalid code

### ISSUE THREAD

**Issue by stephentoub** [dotnet/roslyn#38508]
**Version Used**: 
3.4.0-beta1-19426-03+b18d90790da95b6862def56802fa4249e4868bb4

**Steps to Reproduce**:
Add this code:
```C#
#nullable enable
class Program
{
    static void Main() { }

    static void M()
    {
        object? o;

    }
}
```
Then type:
`o = new`
and hit space.  It pops up IntelliSense to complete the new expression:
![image](https://user-images.githubusercontent.com/2642209/64354381-54d4fa80-cfcd-11e9-9aa8-21eeef8ece3e.png)
Then type Tab.

**Expected Behavior**:
IntelliSense should select `object`, not `object?`.

**Actual Behavior**:
The selected item is `object?`, which is invalid in a constructor:
![image](https://user-images.githubusercontent.com/2642209/64354472-81891200-cfcd-11e9-8a90-51742ab557ed.png)

**Comment by sharwell** [comment]
@ryzngard I wasn't able to reproduce this in 16.3 Preview 3, but I also wasn't able to find the pull request or duplicate issue describing where it would have been fixed. Can you verify and try to find the resolving issue/PR?

**Comment by stephentoub** [comment]
@sharwell, I'm on 16.4.0 Preview 1.0 [29306.174.master], and I still see this.

**Comment by sharwell** [comment]
@stephentoub Ah, looks like it was added later then. I read your version originally as 3.3 beta 1 (16.3 Preview 1), but it's actually a 16.4 branch.

**Comment by davidwengier** [comment]
Ran into this today too on 29310.97
![image](https://user-images.githubusercontent.com/754264/64678205-78f87600-d4bc-11e9-811b-a2bcc8dd3c6d.png)

**Comment by sharwell** [comment]
@jinujoseph This is causing problems for dogfood users. I bumped it up to 16.4 Preview 1 but the earlier we deliver a fix the better it will be for partner teams testing NRT.

**Comment by jasonmalinowski** [comment]
Not sure how this would have broken, but good find @stephentoub. Not sure what would have broken this -- maybe something around target type completion?

**Comment by ryzngard** [comment]
I'm taking a look at this

**Comment by jinujoseph** [comment]
resolved with #38645
